### DISCUSSION TITLE
Getting Warning: Prop `disabled` did not match. Server: "null" Client: "true" for button

### DISCUSSION THREAD

**Discussion by niwsa** [vercel/next.js#17254] [Help]
I have a button with disabled set from state
`<button
            type="button"
            className={styles.left}
            onClick={() => setCurrentPage(currentPage - 1)}
            disabled={currentPage === 0}
          >
            {"<"}
          </button>`
which throws the above warning.Any pointers why this might be happening?

**Comment by TheJarX** [comment]
Did you solve it? I'm getting the same warning

**Reply by niwsa** [reply]
@TheJarX I removed disabled attribute and ended up using className 
```
            <button type="button" className={`btnRst ${
                currentPage === 0 ? styles.cur : ""
              }`} onClick={() => setCurrentPage(0)}></button>
```
**EDIT:** Here the implementation slightly changed instead of using disabled using CSS to show the current active  
button
Something related to the hydration mismatch https://github.com/vercel/next.js/issues/11600#issuecomment-722433356

**Reply by maksym3d** [reply]
After some trial and error, setting 'disabled' programmatically seem to have cleared the error:

<button onClick={() => gotoPage(0)} {...({disabled:!canPreviousPage})}>

**Comment by niwsa** [comment]
I think the real answer can be found here..explained nicely..
https://github.com/vercel/next.js/discussions/17443

**Comment by kkntl** [comment]
Probably the same problem as https://github.com/vercel/next.js/discussions/21999
